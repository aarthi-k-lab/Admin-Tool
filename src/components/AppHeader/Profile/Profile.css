.container {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
  height: 8em;
  justify-content: space-between;
  padding: 1em;
}

.field-title.field-title {
  font-weight: 700;
  width: 5em;
}

.group-list {
  margin: 0;
}

.row {
  display: flex;
  flex-wrap: nowrap;
}
