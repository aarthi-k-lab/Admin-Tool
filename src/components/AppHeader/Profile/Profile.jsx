import React from 'react';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import './Profile.css';
import PropTypes from 'prop-types';
import * as R from 'ramda';

function Profile({ userDetails, groups, skills }) {
  const getEmail = R.propOr('', 'email');
  const getName = R.propOr('', 'name');

  const email = getEmail(userDetails);
  const name = getName(userDetails);

  return (
    <Paper styleName="container" tabIndex={-1}>
      <Typography variant="title">User Profile</Typography>
      <div styleName="row">
        <Typography styleName="field-title" variant="body1">Name: </Typography>
        <Typography variant="body1">{name}</Typography>
      </div>
      <div styleName="row">
        <Typography styleName="field-title" variant="body1">Email: </Typography>
        <Typography variant="body1">{email}</Typography>
      </div>
      <div styleName="row">
        <Typography styleName="field-title" variant="body1">Groups: </Typography>
        <ol styleName="group-list">{Profile.renderGroups(groups)}</ol>
      </div>
      <div styleName="row">
        <Typography styleName="field-title" variant="body1">Skills: </Typography>
        <ol styleName="group-list">{Profile.renderSkills(skills)}</ol>
      </div>
    </Paper>
  );
}

Profile.renderGroups = function renderGroups(groups) {
  return groups.map(
    group => (
      <li key={group}>
        <Typography variant="body1">{group}</Typography>
      </li>
    ),
  );
};

Profile.renderSkills = function renderSkills(skills) {
  return skills && skills.map(
    skill => (
      <li key={skill}>
        <Typography variant="body1">{skill}</Typography>
      </li>
    ),
  );
};

Profile.propTypes = {
  groups: PropTypes.arrayOf(PropTypes.string).isRequired,
  skills: PropTypes.arrayOf(PropTypes.string).isRequired,
  userDetails: PropTypes.shape({
    email: PropTypes.string,
    name: PropTypes.string,
  }).isRequired,
};

export default Profile;
