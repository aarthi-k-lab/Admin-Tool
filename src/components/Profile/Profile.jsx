import React from 'react';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import './Profile.css';
import Auth from '../../lib/Auth';

function Profile() {
  const auth = Auth.getInstance();
  const { email, name } = auth.getUserDetails();
  const groups = auth.getGroups();
  return (
    <Paper styleName="container">
      <Typography variant="title">User Profile</Typography>
      <div styleName="row">
        <Typography styleName="field-title" variant="body1">Email: </Typography>
        <Typography variant="body1">{email}</Typography>
      </div>
      <div styleName="row">
        <Typography styleName="field-title" variant="body1">Groups: </Typography>
        <ol styleName="group-list">{Profile.renderGroups(groups)}</ol>
      </div>
      <div styleName="row">
        <Typography styleName="field-title" variant="body1">Name: </Typography>
        <Typography variant="body1">{name}</Typography>
      </div>
    </Paper>
  );
}

Profile.renderGroups = function renderGroups(groups) {
  return groups.map(
    group => (
      <li key={group}>
        <Typography variant="body1">{group}</Typography>
      </li>
    ),
  );
};

export default Profile;
