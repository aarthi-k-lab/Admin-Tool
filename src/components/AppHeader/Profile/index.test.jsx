import React from 'react';
import renderer from 'react-test-renderer';
import Profile from '.';

describe('<Profile />', () => {
  test('snapshot test', () => {
    const setRoleCallBack = jest.fn();
    const tree = renderer
      .create(
        <Profile
          groups={['group1', 'group2']}
          setRoleCallBack={setRoleCallBack}
          skills={{ FEUW: ['Skill14::InVESTOR', 'Skill19::MRC Onshore', 'Skill1::something', 'Skill2::nothing'] }}
          userDetails={{
            email: 'email',
            name: 'name',
          }}
        />,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
