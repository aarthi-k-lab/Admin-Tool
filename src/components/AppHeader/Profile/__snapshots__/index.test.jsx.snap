// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> snapshot test 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-elevation2-5 MuiPaper-rounded-2 components-AppHeader-Profile-Profile__container--3sj5b"
  tabIndex={-1}
>
  <h2
    className="MuiTypography-root-28 MuiTypography-title-34"
  >
    User Profile
  </h2>
  <div
    className="components-AppHeader-Profile-Profile__row--3EfAD"
  >
    <p
      className="MuiTypography-root-28 MuiTypography-body1-37 components-AppHeader-Profile-Profile__field-title--25ARv"
    >
      Name: 
    </p>
    <p
      className="MuiTypography-root-28 MuiTypography-body1-37"
    >
      name
    </p>
  </div>
  <div
    className="components-AppHeader-Profile-Profile__row--3EfAD"
  >
    <p
      className="MuiTypography-root-28 MuiTypography-body1-37 components-AppHeader-Profile-Profile__field-title--25ARv"
    >
      Email: 
    </p>
    <p
      className="MuiTypography-root-28 MuiTypography-body1-37"
    >
      email
    </p>
  </div>
  <div
    className="components-AppHeader-Profile-Profile__row--3EfAD"
  >
    <p
      className="MuiTypography-root-28 MuiTypography-body1-37 components-AppHeader-Profile-Profile__field-title--25ARv"
    >
      Groups: 
    </p>
    <ol
      className="components-AppHeader-Profile-Profile__group-list--1vsZF"
    >
      <li>
        <p
          className="MuiTypography-root-28 MuiTypography-body1-37"
        >
          group1
        </p>
      </li>
      <li>
        <p
          className="MuiTypography-root-28 MuiTypography-body1-37"
        >
          group2
        </p>
      </li>
    </ol>
  </div>
  <div
    className="components-AppHeader-Profile-Profile__row--3EfAD"
  >
    <p
      className="MuiTypography-root-28 MuiTypography-body1-37 components-AppHeader-Profile-Profile__field-title--25ARv"
    >
      Skills: 
    </p>
    <ol
      className="components-AppHeader-Profile-Profile__group-list--1vsZF"
    >
      <li>
        <p
          className="MuiTypography-root-28 MuiTypography-body1-37"
        >
          0 - Skill14
        </p>
      </li>
      <li>
        <p
          className="MuiTypography-root-28 MuiTypography-body1-37"
        >
          1 - Skill19
        </p>
      </li>
      <li>
        <p
          className="MuiTypography-root-28 MuiTypography-body1-37"
        >
          2 - Skill1
        </p>
      </li>
      <li>
        <p
          className="MuiTypography-root-28 MuiTypography-body1-37"
        >
          3 - Skill2
        </p>
      </li>
    </ol>
  </div>
</div>
`;
